elide-core/src/main/java/com/yahoo/elide/core/filter/Operator.java

```diff
@@ -169,20 +185,21 @@ public static Operator fromString(final String string) {
         };
     }
 
-    private static <T> Predicate<T> notIn(String field, List<Object> values, RequestScope requestScope) {
+    private static <T> Predicate<T> in(String field, List<Object> values,
+                                       RequestScope requestScope, Function<String, String> transform) {
         return (T entity) -> {
-            Object val = getFieldValue(entity, field, requestScope);
+            Class type = requestScope.getDictionary().getType(entity, field);
+            if (!type.isAssignableFrom(String.class)) {
+                throw new IllegalStateException("Cannot case insensitive compare non-string values");
+            }
 
-            return val == null || values.stream()
-                    .map(v -> CoerceUtil.coerce(v, val.getClass()))
-                    .noneMatch(val::equals);
+            String val = transform.apply((String) getFieldValue(entity, field, requestScope));
```

Comment:
I'm wondering if it is a good idea to assume everything must be string assignable for the in operator.

Reply:
I just followed the convention elsewhere in this file to have a specialized version of the function share the same name. In reality this `in` is only called for case insensitive in operations. I'll add clarifying comments though